### PR TITLE
Implement state filtering for elections in the backend DataServlet and frontend directory AngularJS component. Show filtering option when user lists all elections.

### PR DESCRIPTION
**Pull request by KathyFeiyang**
### What is a quick description of the change?

1. Write unit tests for the state filtering functionality in the `DataServlet`.

2. Implement state filtering in the `DataServlet`, handling edge cases where the user has not selected a filter criteria and where the election does not belong to a specific state.

3. Add dropdown menu to the AngularJS directory page template. Data is refreshed at changes in dropdown menu selection, without having to reload the page.

4. Extract user input and pass it to the backend in the AngularJS directory page component, handling edge cases where the user has not selected a filter criteria and where the user has selected no-filtering.

5. _[Added functionality]_ Enable the state filtering dropdown box when the user lists all elections. Disable and hide the dropdown box when the user is already searching for a specific residential address.

### Is this fixing an issue?

none

### Are there more details that are relevant?

none

### Check lists (check `x` in `[ ]` of list items)

- [x] Test written/updating
- [x] Tests passing
- [x] Coding style (indentation, etc)

Please explain why any are not present, if any.

### CODE REVIEW

File: project/src/main/webapp/directory-page/directory-page.component.js [hunk 1]
```diff
@@ -15,11 +15,64 @@
...
+    .component('directoryPage', {
+      templateUrl: '/directory-page/directory-page.template.html', 
+      controller: function directoryPageController($scope) {
+        showOrHideStateFilter($scope);
+        getElectionCandidateInformation($scope);
+        $scope.filterOnState = function() {
+          getElectionCandidateInformation($scope);
+        }
+        $scope.states =
+            ['--', 'AL', 'AK', 'AR', 'AZ', 'CA', 'CO', 'CT', 'DE', 'FL', 'GA',
```
Comment (suparnabarua) [inline project/src/main/webapp/directory-page/directory-page.component.js:28]: Why the --?

File: project/src/main/webapp/style.css [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+#navigation-bar a {
+  color: #474747;
+  font-family: 'Roboto Slab', serif;
+}
+
+#site-header {
+  color: #f0eded;
+  background-color: #b32424;
+  font-family: 'Roboto Slab', serif;
+}
```
Comment (suparnabarua) [inline project/src/main/webapp/style.css]: nit new line

File: project/src/main/webapp/navigation/nav-bar.template.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+      <a class="nav-link" href="/index.html">Home</a>
+    </li>
+    <li class="navbar-item">
+      <a class="nav-link" href="/education.html">Voter Education</a>
+    </li>
+    <li class="navbar-item">
+      <a class="nav-link" href="/">Polling Locations</a>
+    </li>
+  </ul>
+</nav>
```
Comment (suparnabarua) [inline project/src/main/webapp/navigation/nav-bar.template.html]: nit new line

**Review by suparnabarua — CHANGES_REQUESTED** [review]

File: project/src/main/webapp/directory-page/directory-page.component.js (hunk 1 above)
Comment (KathyFeiyang) [reply project/src/main/webapp/directory-page/directory-page.component.js:28]: This is for when the user decides to go back to viewing all elections. The dropdown box doesn't automatically provide a no-selection option after you selected one.

**Review by suparnabarua — APPROVED** [review]

**Review by kellyktan — APPROVED** [review]

**Review by tjyoung99 — APPROVED** [review]

**Review by ganewman — APPROVED** [review]
